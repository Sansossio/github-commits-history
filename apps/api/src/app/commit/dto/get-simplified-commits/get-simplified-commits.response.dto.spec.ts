import { GetSimplifiedCommitsResponseDTO } from './get-simplified-commits.response.dto'

describe('GetSimplifiedCommitsResponseDTO', () => {
  it('should be defined', () => {
    expect(GetSimplifiedCommitsResponseDTO).toBeDefined()
  })

  it('should match snapshot using commit.commiter.date', () => {
    const obj: any = {
      sha: 'sha',
      commit: {
        message: 'test message',
        committer: {
          date: new Date(2022, 10, 7, 1, 0, 0, 0)
        },
      },
      author: {},
      committer: {},
      url: 'https://test.com'
    }
    expect(GetSimplifiedCommitsResponseDTO.fromGithubResponse(obj)).toMatchSnapshot()
  })

  it('should match snapshot using commit.author.date', () => {
    const obj: any = {
      sha: 'sha',
      commit: {
        message: 'test message',
        committer: {},
        author: {
          date: new Date(2021, 10, 7, 1, 0, 0, 0)
        }
      },
      author: {},
      committer: {},
      url: 'https://test.com'
    }
    expect(GetSimplifiedCommitsResponseDTO.fromGithubResponse(obj)).toMatchSnapshot()
  })
})
