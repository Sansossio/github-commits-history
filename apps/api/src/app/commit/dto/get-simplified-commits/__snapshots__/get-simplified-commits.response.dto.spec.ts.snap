// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSimplifiedCommitsResponseDTO should match snapshot using commit.author.date 1`] = `
GetSimplifiedCommitsResponseDTO {
  "author": Object {
    "avatar_url": undefined,
    "id": undefined,
    "login": undefined,
    "url": undefined,
  },
  "committer": Object {
    "avatar_url": undefined,
    "id": undefined,
    "login": undefined,
    "url": undefined,
  },
  "date": 2021-11-07T05:00:00.000Z,
  "message": "test message",
  "sha": "sha",
  "url": "https://test.com",
}
`;

exports[`GetSimplifiedCommitsResponseDTO should match snapshot using commit.commiter.date 1`] = `
GetSimplifiedCommitsResponseDTO {
  "author": Object {
    "avatar_url": undefined,
    "id": undefined,
    "login": undefined,
    "url": undefined,
  },
  "committer": Object {
    "avatar_url": undefined,
    "id": undefined,
    "login": undefined,
    "url": undefined,
  },
  "date": 2022-11-07T05:00:00.000Z,
  "message": "test message",
  "sha": "sha",
  "url": "https://test.com",
}
`;
